test/test_sparse_csr.py

```diff
@@ -1273,6 +1275,114 @@ def test_sparse_csr_unary_inplace(self, device, dtype, op):
             self.assertEqual(actual.col_indices(), expect.col_indices())
             self.assertEqual(actual._nnz(), expect._nnz())
 
+
+    @unittest.expectedFailure
+    @ops(sparse_csr_unary_ufuncs, dtypes=OpDTypes.supported, allowed_dtypes=[torch.double, torch.cdouble])
+    def test_autograd_sparse_csr_unary(self, device, dtype, op):
+        samples = list(op.sample_inputs(device, dtype))
+
+        # Fail early to prevent silent success with this test
+        ndims_equals_2d = (s.input.ndim == 2 for s in samples)
+        if not any(ndims_equals_2d):
+            raise ValueError("Expected at least one 2D tensor in samples.")
+
+        for sample in samples:
+            sparse_input = sample.input.to_sparse_csr().requires_grad_(True)
+
+            def fn(input):
+                output = op.gradcheck_wrapper(op.get_op(), input, *sample.args, **sample.kwargs)
+                output = output.to_dense()
+                if sample.output_process_fn_grad is not None:
+                    return sample.output_process_fn_grad(output)
+                return output
+
+            # NotImplementedError inside gradcheck when computing numerical Jacobian
```

Comment:
It could be useful to explicitly regex on this. A change here could actually be signal caused by some faulty diff.

Reply:
With regex checks, the status of this test would be changed from XFAIL to PASSED. I think when tests pass they test the intended behavior and nothing needs to be done. xfails are signals to action to actually fix stuff.
xfails are not used much in the PyTorch testing though.

Once the gradcheck begins to work for sparse csr input it would be a signal to revisit this test.